language/diem-framework/modules/Vote.move

```diff
@@ -0,0 +1,377 @@
+    ): WeightedVoter {
+        WeightedVoter {
+            weight,
+            voter,
+        }
+    }
+
+    /// Create a ballot under the signer's address and return the `BallotID`
+    public fun create_ballot<Proposal: store + copy + drop>(
+        ballot_account: &signer,
+        proposal: Proposal,
+        proposal_type: vector<u8>,
+        num_votes_required: u64,
+        allowed_voters: vector<WeightedVoter>,
+        expiration_timestamp_secs: u64
+    ): BallotID acquires Ballots, BallotCounter {
+        let ballot_address = Signer::address_of(ballot_account);
+
+        assert(DiemTimestamp::now_seconds() < expiration_timestamp_secs, Errors::invalid_argument(EINVALID_TIMESTAMP));
+
+        if (!exists<BallotCounter>(ballot_address)) {
+            move_to(ballot_account, BallotCounter {
+                counter: 0,
+            });
+        };
+        if (!exists<Ballots<Proposal>>(ballot_address)) {
+            move_to(ballot_account, Ballots<Proposal> {
+                ballots: Vector::empty(),
+                create_ballot_handle: Event::new_event_handle<CreateBallotEvent<Proposal>>(ballot_account),
+                remove_ballot_handle: Event::new_event_handle<RemoveBallotEvent>(ballot_account),
+                voted_handle: Event::new_event_handle<VotedEvent>(ballot_account),
+                ballot_approved_handle: Event::new_event_handle<BallotApprovedEvent>(ballot_account),
+            });
+        };
+
+        let ballot_data = borrow_global_mut<Ballots<Proposal>>(ballot_address);
+
+        // Remove any expired ballots
+        gc_internal<Proposal>(ballot_data);
+        let ballots = &mut ballot_data.ballots;
+
+        assert(Vector::length(ballots) < MAX_BALLOTS_PER_PROPOSAL_TYPE_PER_ADDRESS, Errors::limit_exceeded(ETOO_MANY_BALLOTS));
+        let ballot_id = new_ballot_id(incr_counter(ballot_account), ballot_address);
+        let ballot = Ballot<Proposal> {
+            proposal,
+            proposal_type,
+            num_votes_required,
+            allowed_voters,
+            votes_received: Vector::empty(),
+            total_weighted_votes_received: 0,
+            ballot_id: *(&ballot_id),
+            expiration_timestamp_secs,
+        };
+        Vector::push_back(ballots, *(&ballot));
+        Event::emit_event<CreateBallotEvent<Proposal>>(
+            &mut borrow_global_mut<Ballots<Proposal>>(ballot_address).create_ballot_handle,
+            CreateBallotEvent {
+                ballot_id: *(&ballot_id),
+                ballot,
+            },
+        );
+        ballot_id
+    }
+
+    // Checks if a voter is present in the vector<WeightedVoter>
+    fun check_voter_present(
+        weighted_voters: &vector<WeightedVoter>,
+        voter: &vector<u8>,
+    ): bool {
+        let i = 0;
+        let len = Vector::length(weighted_voters);
+        while (i < len) {
+            if (&Vector::borrow(weighted_voters, i).voter == voter) {
+                return true
+            };
```

Comment:
Stylistic nit (feel free to ignore if you prefer this way)
```suggestion
            if (&Vector::borrow(weighted_voters, i).voter == voter) return true;
```

Reply:
Done